File: cpp/array/generate_primes.cpp
#include<iostream>
#include<vector>
#include<deque>

using namespace std;
vector<int> generate_primes(int n){
    vector<int> primes;
    deque<bool>is_prime(n+1,true);
    is_prime[0]=false;
    is_prime[1]=false;
    for(int p=2;p<n;p++){
        if (is_prime[p])
        {
            primes.push_back(p);
            for (auto j = p;j<n;j+=j)
            {
                is_prime[j] = false;
            }
            
        }
        
        
    }
return primes;


    
    
}
int main(){
    vector<int>nums={};
    int n =9;
    nums = generate_primes(n);
      for(size_t i =0;i <nums.size();i++){
        cout<<nums[i]<<",";
      }
      cout<<endl;
    return 0;

}

File: python/array/generate_primes.py
def generate_primes (n: int) :
    primes=[]
    is_prime = [False,False] + [True] * (n-1)
    is_prime[0]=0
    is_prime[1]=0
    for p in range(2,n+1):
        if is_prime[p]:
            primes.append(p)
            for i in range(p*2,n+1,p):
                is_prime[i]=False
    return primes

def main():
    print(generate_primes(9))

if __name__ == "__main__":
    main()
